e2e-tests/batch-mode-tests.sh

```diff
@@ -1,119 +1,160 @@
-
-mvn exec:java -pl batch \
- "-Dexec.args=--openmrsServerUrl=http://localhost:8099/openmrs --openmrsUserName=admin --openmrsPassword=Admin123 \
-   --searchList=Patient --batchSize=20 --targetParallelism=20 \
-   --fhirSinkPath=http://localhost:8098/fhir \
-   --sinkUserName=hapi --sinkPassword=hapi"
-cd "${test_dir}"
-mkdir fhir
-echo "${print_prefix} Counting number of patients, encounters and obs sinked to fhir files"
-curl -L -X GET -u hapi:hapi --connect-timeout 5 --max-time 20 \
-      http://localhost:8098/fhir/Patient/?_summary=count 2>/dev/null >> ./fhir/patients.json
-total_patients_sinked_fhir=$(jq '.total' ./fhir/patients.json)
-echo "${print_prefix} Total patients sinked to fhir ---> ${total_patients_sinked_fhir}"
-if [[ ${total_patients_sinked_fhir} == ${total_test_patients} ]] ;
-then
-  echo "${print_prefix} BATCH MODE WITH FHIR SERVER SINK EXECUTED SUCCESSFULLY"
-else
-  echo "${print_prefix} BATCH MODE WITH FHIR SERVER SINK TEST FAILED"
-  exit 1
-fi
+  fi
+  cd "${test_dir}"
+  mkdir omrs
+  echo "${print_prefix} Finding number of patients, encounters and obs in openmrs server"
+  curl -L -X GET -u admin:Admin123 --connect-timeout 5 --max-time 20 \
+    http://localhost:8099/openmrs/ws/fhir2/R4/Patient/ 2>/dev/null >>./omrs/patients.json
+  total_test_patients=$(jq '.total' ./omrs/patients.json)
+  echo "${print_prefix} Total openmrs test patients ---> ${total_test_patients}"
+  curl -L -X GET -u admin:Admin123 --connect-timeout 5 --max-time 20 \
+    http://localhost:8099/openmrs/ws/fhir2/R4/Encounter/ 2>/dev/null >>./omrs/encounters.json
+  total_test_encounters=$(jq '.total' ./omrs/encounters.json)
+  echo "${print_prefix} Total openmrs test encounters ---> ${total_test_encounters}"
+  curl -L -X GET -u admin:Admin123 --connect-timeout 5 --max-time 20 \
+    http://localhost:8099/openmrs/ws/fhir2/R4/Observation/ 2>/dev/null >>./omrs/obs.json
+  total_test_obs=$(jq '.total' ./omrs/obs.json)
+  echo "${print_prefix} Total openmrs test obs ---> ${total_test_obs}"
+  echo "${print_prefix} Counting number of patients, encounters and obs sinked to parquet files"
+  total_patients_streamed=$(java -jar ./parquet-tools-1.11.1.jar rowcount ./Patient/ | awk '{print $3}')
+  echo "${print_prefix} Total patients synced to parquet ---> ${total_patients_streamed}"
+  total_encounters_streamed=$(java -jar ./parquet-tools-1.11.1.jar rowcount ./Encounter/ | awk '{print $3}')
+  echo "${print_prefix} Total encounters synced to parquet ---> ${total_encounters_streamed}"
+  total_obs_streamed=$(java -jar ./parquet-tools-1.11.1.jar rowcount ./Observation/ | awk '{print $3}')
+  echo "${print_prefix} Total obs synced to parquet ---> ${total_obs_streamed}"
+  if [[ ${total_patients_streamed} == ${total_test_patients} && ${total_encounters_streamed} == ${total_test_encounters} && $total_obs_streamed == ${total_test_obs} ]] \
+    ; then
+    echo "${print_prefix} BATCH MODE WITH PARQUET SYNC EXECUTED SUCCESSFULLY USING $mode MODE"
```

Comment:
nit: Try to use `{}` for all your variable usage, e.g.,
```suggestion
    echo "${print_prefix} BATCH MODE WITH PARQUET SYNC EXECUTED SUCCESSFULLY USING ${mode} MODE"
```